packages/style-engine/src/types.ts

```diff
@@ -11,7 +11,31 @@ export type Box< T extends BoxVariants = undefined > = {
 	left?: CSSProperties[ T extends undefined ? 'left' : `${ T }Left` ];
 };
 
+export type BorderIndividualProperty = 'top' | 'right' | 'bottom' | 'left';
+export type BorderIndividualStyles< T extends BorderIndividualProperty > = {
+	color?: CSSProperties[ `border${ Capitalize< string & T > }Color` ];
```

Comment:
My TS fu isn't great. Very glad to hear if there's a fancier way of doing this.

Reply:
My TS fu isn't very good, either, but this looks reasonable to me. I often find these rules a bit challenging to parse visually, but it given the sheer number of rules for each side, this reads better than the duplication of hard-coding each CSS property to me 👍

Reply:
Perhaps a simple inline comment with an example of the results might help people grok this quicker? Also, renaming `T` to something more explanatory could be another option.